pkg/meta/base.go

```diff
@@ -2444,3 +2478,162 @@ LOOP:
 	}
 	return eno
 }
+
+func (m *baseMeta) setAttr(ctx Context, inode Ino, set uint16, cur, attr *Attr, now time.Time) (*Attr, syscall.Errno) {
+	dirtyAttr := *cur
```

Comment:
why not just change the `cur` here?

Reply:
Because we need the fields (`uid` and `gid`) of `cur` to check permission, I think changing them in the mid may cause bugs. We can also rearrange the control flow, however, I think keep the `cur` immutable is a better solution.